### PR TITLE
NEW : Add filter on categorie actiocomm

### PR DESCRIPTION
**Pull request by defrance**
# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "FIX", "CLOSE" or "NEW" section* (use uppercase to have the PR appears into the ChangeLog, lowercase will not appears)
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed texts with meaningful information*

# FIX|Fix #[*issue_number Short description*]
[*Long description*]

# CLOSE|Close #[*issue_number Short description*]
[*Long description*]

# NEW|New [*Short description*]
[*Long description*]

### CODE REVIEW

File: htdocs/comm/action/index.php [hunk 1]
```diff
@@ -683,6 +687,11 @@
 }
 $sql .= ' WHERE a.fk_action = ca.id';
 $sql .= ' AND a.entity IN ('.getEntity('agenda').')';
+
+if ($search_category > 0) {
+	$sql .= " AND cac.fk_categorie = ".$search_category;
```
Comment (frederic34) [inline htdocs/comm/action/index.php:692]: @defrance add `(int)` before $search_category to make travis happy

File: htdocs/comm/action/list.php [hunk 2]
```diff
@@ -451,6 +455,9 @@
 }
 $sql .= " WHERE c.id = a.fk_action";
 $sql .= ' AND a.entity IN ('.getEntity('agenda').')';
+if ($search_category > 0) {
+	$sql .= " AND cac.fk_categorie = ".$search_category;
```
Comment (frederic34) [inline htdocs/comm/action/list.php:459]: @defrance add `(int)` before $search_category to make travis happy again

**Comment by eldy** [comment]
Search/filter on categories must now be implemented using the sql EXISTS and NOT EXISTS (so we can search on several categories, search on no category defined, and filter can be added into the WHERE only with no impact on the SELECT and the FROM, so this is compatible with a future advanced filter tool).
See examples into projet/list.php

File: htdocs/comm/action/list.php [hunk 3]
```diff
@@ -441,6 +443,9 @@
 if (!$user->hasRight('societe', 'client', 'voir') && !$socid) {
 	$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON a.fk_soc = sc.fk_soc";
 }
+if ($search_category > 0) {
```
Comment (eldy) [inline htdocs/comm/action/list.php:446]: Filter on category must no more be managed by adding a LEFT JOIN.
Nothing must be added in the select and nothing into the FROM. addition must be done into the WHERE only, like done in order or proposal list in v19 (so we can filter on several categories, and also on "no category"). Example in proposal list.php

```
// Search for tag/category ($searchCategoryCustomerList is an array of ID)
$searchCategoryCustomerOperator = -1;
$searchCategoryCustomerList = array($search_categ_cus);
if (!empty($searchCategoryCustomerList)) {
	$searchCategoryCustomerSqlList = array();
	$listofcategoryid = '';
	foreach ($searchCategoryCustomerList as $searchCategoryCustomer) {
		if (intval($searchCategoryCustomer) == -2) {
			$searchCategoryCustomerSqlList[] = "NOT EXISTS (SELECT cs.fk_soc FROM ".MAIN_DB_PREFIX."categorie_societe as cs WHERE s.rowid = cs.fk_soc)";
		} elseif (intval($searchCategoryCustomer) > 0) {
			if ($searchCategoryCustomerOperator == 0) {
				$searchCategoryCustomerSqlList[] = " EXISTS (SELECT cs.fk_soc FROM ".MAIN_DB_PREFIX."categorie_societe as cs WHERE s.rowid = cs.fk_soc AND cs.fk_categorie = ".((int) $searchCategoryCustomer).")";
			} else {
				$listofcategoryid .= ($listofcategoryid ? ', ' : '') .((int) $searchCategoryCustomer);
			}
		}
	}
	if ($listofcategoryid) {
		$searchCategoryCustomerSqlList[] = " EXISTS (SELECT cs.fk_soc FROM ".MAIN_DB_PREFIX."categorie_societe as cs WHERE s.rowid = cs.fk_soc AND cs.fk_categorie IN (".$db->sanitize($listofcategoryid)."))";
	}
	if ($searchCategoryCustomerOperator == 1) {
		if (!empty($searchCategoryCustomerSqlList)) {
			$sql .= " AND (".implode(' OR ', $searchCategoryCustomerSqlList).")";
		}
	} else {
		if (!empty($searchCategoryCustomerSqlList)) {
			$sql .= " AND (".implode(' AND ', $searchCategoryCustomerSqlList).")";
		}
	}
}
```

**Comment by eldy** [comment]
It seems this PR was set with "Discussion" or "PR to fix" tag, but no answer nor correction was provided since a long time. So request is automatically closed.
Please reopened and add a comment if you think this is an error.

**Pull request closed without merging**